### DISCUSSION TITLE
Whisper struggling with transcribing some YouTube videos

### DISCUSSION THREAD

**Discussion by qaixerabbas** [openai/whisper#1494] [Q&A] [answered]
I have used Whisper for transcription few weeks before and it worked well even for long videos. But now when I am trying to transcribe this video https://www.youtube.com/watch?v=KqbXZJ80yFU it results in hallucinations. The transcription makes no sense. I have tried to trim the file, set the required sampling rate (16000) and converted the file to mono wav as well but no luck so far. I am not sure what is wrong with this file. I even removed the first few seconds where there is no speech or sound.

I tried this with other videos from YouTube as well and its struggling with the transcription. I have tried base, small, medium and large models as well. The input video is clear and audible but still I am not sure what is wrong with the file.

Here is the transcription generated by Large Whisper model (file trimmed to first 3 minutes)

> "Helo Jodie, diolch am i chi eich gweld eich newydd. Sut eich bod chi ond oddi wrth i mi gweld chi'r diwedd? Ie, rwy'n bwysig. Rwy'n cydweithio ar y plan, felly mae'n mynd iawn. Oh, y plan a wnaeth y hosbital eich seilio? Ie. Ie, dwi'n ffantastig. A sut rydych chi wedi'i ddod o hyd i hynny? Ie, mae'n bod yn dda. Rwy'n ceisio ei dilyn, ond yn yr wythnos diwethaf, nid oedd yn dda iawn. Roedd gen i ddau fath o dydd anodd, ond o'r dydd hwn, rwy'n ddod o hyd. Ie, felly dwi'n meddwl am y dydd anodd hwn. Beth y gwnaethoch chi'n ei chwyno ar y dydd hon? Dwi ddim yn meddwl bod gen i unrhyw beth. Ie, felly dydych chi ddim yn gwneud unrhyw beth. Ie. Ie, dwi'n meddwl am y dydd hon. Pam oedd y dydd yn anodd? A oedd unrhyw beth yn digwydd o'r amser hon? Mae'n bod yn anodd iawn ar ysgol, oherwydd mae gennym ein hysgafon, ac roedd gennym ysgafn y dydd nesaf, felly rwy'n meddwl roeddwn i'n bwysig am hynny, felly roeddwn i'n ceisio cymryd sylw ar hynny yn ystod hyn. Felly roedd gennych chi'n teimlo'n anhygoel am yr hysgafon sy'n dod i mewn. A oedd unrhyw beth eraill sy'n eich bwysig? Dwi'n meddwl roedd gennym, roedd gennym ddysgwyr ffordd y 6. Ac roedd hynny'n dod ymlaen yn y dyddiau desaf, ac roeddwn i ddim eisiau mynd i ddysgu am hynny. Felly roedd yna rhai pethau sy'n eich bwysig, sydd ar eich meddwl. Pa beth yw'n eich bwysig mwy, dwi'n meddwl? Efallai'r ffordd y 6. Ie, y ffordd y 6. Iawn. Pa mor oedd eich bod chi'n teimlo am hynny, pan ddewchon ni'r sgwrs am hynny nawr, pan mae hynny'n cael ei ddod i mewn, pa ffordd o emociau y dywedwch chi? Dwi ddim eisiau dechrau y ffordd y 6. Mae'n eithaf yn ddiweddar, ond dwi ddim eisiau. Dwi'n meddwl, dwi'n cael ysgafnion ymlaen, felly dwi'n cael meddwl am hynny, ond dwi ddim eisiau mynd i'r ffordd y 6. Yn ystod y blynyddoedd. Iawn, felly, beth yw'r ffordd ym mhobol ffordd sy'n gwneud i chi ddim eisiau mynd? Rhaid i mi ddechrau arno newydd a chadw ffrindiau newydd a newydd ffrindiau newydd a pethau, ac nid wyf yn teimlo'n barod i hynny. Beth ydych chi'n meddwl ei fod yn digwydd pan fyddwch yn rhaid i chi gadw ffrindiau newydd a dechrau newydd ffrindiau newydd? Dwi ddim, dwi ddim yn gwybod os ydw i'n mynd i fod yn hoffi i mi, fel, os ydynt ddim eisiau siarad â fi, ac yna dwi ddim yn gallu sefydlu ar fy swydd a chadw hwnnw. Felly dydych chi'n beryglu efallai nad ydych chi'n gallu gadw ffrindiau? Ie. A ydych chi'n beryglu efallai nad ydych chi'n gallu sefydlu ar fwrdd? Ie. Ac sut ydych chi'n teimlo pan fyddwch chi'n cael y penderfyniadau hynny? Ym, ychydig yn beryglu. Dwi ddim yn gwybod beth y bydd pobl yn meddwl, felly dwi ddim eisiau eu hyfforddi a dweud unrhyw beth yn iawn neu gwneud unrhyw beth yn iawn. "

Any help in this regard is highly appreciated.

**Comment by phineas-pta** [comment]
whisper misdetect the language

u must specify `language="english"`

**Reply by qaixerabbas** [reply]
@phineas-pta thank you so much for your comment. When I do not specify language for other audios, I get perfect transcriptions. But for this video (and videos like this) why I have to specify the language? I want to know is there any underlying accent issues with the model. If you have any information please let me know.

P.S: Specifying language worked fine for transcriptions and even a base model was able to correctly transcribe the video.

**Reply by phineas-pta** [reply, marked as answer]
as i said: whisper misdetect the language

for better transcriptions, u should always specify language
